### PR TITLE
Better doco for PorousFlowMassFraction

### LINKED ISSUE #17297: Typo in PorousFlowMassFraction doco
## Reason

The doco string for `mass_fraction_vars` is wrong.  It should have `N-2` in most places where `N-1` appears

## Design

correct it!

## Impact

reduce user confusion

### PR DESCRIPTION
**Pull request by WilkAndy**
Fixes #17297

### CODE REVIEW

File: modules/porous_flow/doc/content/source/materials/PorousFlowMassFraction.md [hunk 1]
```diff
@@ -2,6 +2,33 @@
 
 !syntax description /Materials/PorousFlowMassFraction
 
+The `mass_fraction_vars` is a list of mass fractions, $\chi_{\beta}^{\kappa}$, of fluid component $\kappa$ in phase $\beta$:
```
Comment (cpgr) [inline modules/porous_flow/doc/content/source/materials/PorousFlowMassFraction.md:5]: This is much better for users. Thanks Andy

**Review by cpgr — APPROVED** [review]

File: modules/porous_flow/doc/content/source/materials/PorousFlowMassFraction.md (hunk 1 above)
Comment (WilkAndy) [reply modules/porous_flow/doc/content/source/materials/PorousFlowMassFraction.md:5]: Thanks for the review @cpgr .    The reason for this change is i got confused when helping Scott H while you were away.  I couldn't remember which mass fractions we had to specify!

**Review by loganharbour — APPROVED** [review]
Will wait for a ✅ from @cpgr

**Review by cpgr — APPROVED** [review]
👍

**Pull request merged**